### PR TITLE
Machine control docs

### LINKED ISSUE #43: SCP Documentation partly missing
* [x] Doc comments
* [ ] Introductory overview
* [ ] Selection of worked examples
* [ ] Basic API docs in Sphinx

### PR DESCRIPTION
**Pull request by mossblaser**
Fixes #43

### CODE REVIEW

File: rig/machine_control/machine_controller.py [hunk 1]
```diff
@@ -106,33 +110,46 @@ def __init__(self, initial_host, scp_port=consts.SCP_PORT,
...
-        """Transmit an SCP Packet.
+        """Transmit an SCP Packet and return the response.
 
-        See the arguments for
-        :py:method:`~rig.machine_control.scp_connection.SCPConnection` for
-        details.
+        This function is a thin wrapper around
+        :py:meth:`~rig.machine_control.scp_connection.SCPConnection`.
+
+        Future versions of this command will automatically chose the most
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: "choose"

File: rig/machine_control/machine_controller.py [hunk 2]
```diff
@@ -186,9 +220,11 @@ def boot(self, width, height, **boot_kwargs):
         The constants `rig.machine_control.boot.spinX_boot_options` can be used
         to specify boot parameters, for example::
 
-            controller.boot(**spin5_boot_options)
+            controller.boot(**spine_boot_options)
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: "spine"?

File: rig/machine_control/machine_controller.py [hunk 3]
```diff
@@ -544,12 +613,12 @@ def sdram_alloc(self, size, tag=0, x=Required, y=Required,
     @ContextMixin.use_contextual_arguments
     def sdram_alloc_as_io(self, size, tag=0, x=Required, y=Required,
                           app_id=Required):
-        """Like :py:meth:`.sdram_alloc` but a file-like object which allows
-        reading and writing to the region is returned.
+        """Like :py:meth:`.sdram_alloc` but returns a file-like object which
+        allows safe reading and writing to the block is allocated.
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: "block that is"

File: rig/machine_control/machine_controller.py [hunk 4]
```diff
@@ -834,9 +904,14 @@ def send_signal(self, signal, app_id=Required):
     def load_routing_tables(self, routing_tables, app_id=Required):
         """Allocate space for an load multicast routing tables.
 
+        The routing table entries will be removed automatically when the
+        associated application is stopped.
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: This doesn't appear to happen and I haven't chased down why.  It should though...

File: rig/machine_control/machine_controller.py [hunk 5]
```diff
@@ -1215,12 +1304,14 @@ def __str__(self):
 
 
 class MemoryIO(object):
+    """A file-like view into the memory-space of a chip."""
```
Comment (mundya) [inline rig/machine_control/machine_controller.py]: "A file-like view into a subspace of the memory-space of a chip."?

**Comment by mundya** [comment]
L(V)GTM!  Thank you for doing this :)

File: rig/machine_control/machine_controller.py (hunk 4 above)
Comment (mossblaser) [inline rig/machine_control/machine_controller.py]: Opened #67

**Comment by mossblaser** [comment]
Thank you @mundya! Please merge if happy!

**Pull request merged**